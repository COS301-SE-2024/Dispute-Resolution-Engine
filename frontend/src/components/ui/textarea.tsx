import * as React from "react";

import { cn } from "@/lib/utils";

export interface TextareaProps extends React.TextareaHTMLAttributes<HTMLTextAreaElement> {}

const Textarea = React.forwardRef<HTMLTextAreaElement, TextareaProps>(
  ({ className, ...props }, ref) => {
    return (
      <textarea
        className={cn(
          "flex w-full rounded-md border py-4 px-3 border-zinc-200 bg-white  text-sm ring-offset-white placeholder:text-zinc-500 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-zinc-950 focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 dark:border-dre-100/20 dark:bg-dre-bg-dark dark:ring-offset-zinc-950 dark:placeholder:text-white/50 dark:focus-visible:ring-zinc-300",
          className
        )}
        ref={ref}
        {...props}
      />
    );
  }
);
Textarea.displayName = "Textarea";

export { Textarea };
