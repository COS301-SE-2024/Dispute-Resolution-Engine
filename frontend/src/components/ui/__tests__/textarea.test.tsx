import renderer from 'react-test-renderer';
import { Textarea } from '../textarea';

it('renders correctly', () => {
  const tree = renderer
    .create(<Textarea />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
