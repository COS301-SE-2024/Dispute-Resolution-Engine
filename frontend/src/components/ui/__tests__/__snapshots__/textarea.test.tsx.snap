// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<textarea
  className="flex min-h-[80px] w-full rounded-md border border-primary-500/30 bg-surface-light-100 px-3 py-2 text-sm placeholder:text-slate-500 focus-visible:bg-white focus-visible:outline-none focus-visible:ring-2 ring-primary-500 disabled:cursor-not-allowed disabled:text-black/50 dark:disabled:text-white/50 dark:border-primary-500/30 dark:bg-surface-dark-900 dark:placeholder:text-slate-400 dark:focus-visible:bg-surface-dark-800 transition-colors"
/>
`;
